sparkle/src/components/SearchInput.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { forwardRef } from "react";
+
+import { Button, Icon, Input } from "@sparkle/components";
+import { MagnifyingGlassIcon, XMarkIcon } from "@sparkle/icons";
+import { cn } from "@sparkle/lib/utils";
+
+export interface SearchInputProps {
+  placeholder?: string;
+  value: string | null;
+  onChange?: (value: string) => void;
```

Comment:
OOC: are there cases where we don't want an `onChange` at all? I feel like it would be weird to not have one that at least updates the `value`.

Reply:
Fair point, I guess when disabled? But don't really see a use case for now so let me make it required 😇